### ISSUE TITLE
[bug] ExpressionEvaluator peeks empty stack

### ISSUE THREAD

**Issue by lofcz** [codingseb/ExpressionEvaluator#70]
Hi, this is a snippet from latest version of EE, starting from line 2942. 
If stack count is eq to 0 you are calling `stack.Peek()`, which results in an unthrown error. The comment on this line `else if (evaluationStackCount == 1)` is your original code, commented by me.

```csharp
stack.Clear();
for (int i = 0; i < list.Count; i++)
{
   stack.Push(list[i]);
}

if (stack.Count > 1)
{
    foreach (var item in stack)
    {
        if (item is BubbleExceptionContainer bubbleExceptionContainer)
        {
            throw bubbleExceptionContainer.Exception; //Throw the first occuring error
        }
    }
    throw new ExpressionEvaluatorSyntaxErrorException("Syntax error. Check that no operator is missing");
}
else if (evaluationStackCount == 1) //  && stack.Peek() is BubbleExceptionContainer bubbleExceptionContainer <-- peek on empty stack
{
    //We reached the top level of the evaluation. So we want to throw the resulting exception.
    throw new Exception(); // <-- here you throw bubbleExceptionContainer in original code
}

return stack.Pop();
```

Best possible resolution here would be to properly throw why the error occured. 

Occurs on zero tokens eval: 
```csharp
new EE().Evaluate("")
```

**Comment by lofcz** [comment]
Removing `stack.Clear();` on line 2942 should be enough to fix this, is there any specific reason to have it there?

**Comment by codingseb** [comment]
Yes Stack Empty is not the best exception message in this case.

I added the following code in the top of  the method `ProcessStack(Stack<object> stack)` :  
```c#
if (stack.Count == 0)
    throw new ExpressionEvaluatorSyntaxErrorException("Empty expression or no token found");
```

I published version 1.4.18.0 with this.

**Comment by lofcz** [comment]
Great, thanks. Really looking for v2, thanks foir doing this!
